flink-table/flink-table-planner/src/test/java/org/apache/flink/connector/source/split/ValuesSourcePartitionSplitSerializer.java

```diff
@@ -68,7 +70,11 @@ public ValuesSourcePartitionSplit deserialize(int version, byte[] serialized)
                 String value = in.readUTF();
                 partition.put(key, value);
             }
-            ValuesSourcePartitionSplit split = new ValuesSourcePartitionSplit(partition);
+            final boolean isInfinite = in.readBoolean();
+            ValuesSourcePartitionSplit split =
+                    new ValuesSourcePartitionSplit(
+                            partition,
+                            isInfinite ? FinishingLogic.INFINITE : FinishingLogic.FINITE);
```

Comment:
We have a lot of these if checks. Maybe introduce a method in `FinishingLogic` serialization and deserialization?